src/PrestaShopBundle/Entity/ApiAccess.php

```diff
@@ -30,60 +30,73 @@
 
 use Doctrine\ORM\Mapping as ORM;
 use PrestaShop\PrestaShop\Core\Domain\AuthorizationServer\Model\ApiAccessInterface;
-use PrestaShop\PrestaShop\Core\Domain\AuthorizationServer\Model\AuthorizedApplicationInterface;
 use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
+use Symfony\Component\Validator\Constraints as Assert;
 
 /**
  * @ORM\Entity(repositoryClass="PrestaShopBundle\Entity\Repository\ApiAccessRepository")
  * @ORM\Table()
- * @UniqueEntity("name")
  *
  * @experimental
  */
+#[UniqueEntity('clientId')]
+#[UniqueEntity('clientName')]
 class ApiAccess implements ApiAccessInterface
 {
     /**
-     * @var int
-     *
      * @ORM\Id
-     * @ORM\Column(name="id_api_access", type="integer", options={"unsigned":true})
+     * @ORM\Column(name="id_api_access", type="integer", options={"unsigned": true})
      * @ORM\GeneratedValue(strategy="AUTO")
      */
-    private $id;
+    #[Assert\Type('integer')]
+    #[Assert\Positive]
+    private int $id;
 
     /**
-     * @var string
-     *
      * @ORM\Column(name="client_id", type="string", length=255)
      */
-    private $clientId;
+    #[Assert\Type('string')]
+    #[Assert\Length(max: 255, options: ['allowEmptyString' => false])]
+    #[Assert\NotBlank]
```

Comment:
So even with the `allowEmptyString` the `NotBlank` is still needed, then what's the point of `allowEmptyString` ? 🤔